src/main/java/com/expercise/interpreter/typechecker/javascript/ListJavaTypeWrapper.java

```diff
@@ -0,0 +1,20 @@
+package com.expercise.interpreter.typechecker.javascript;
+
+import jdk.nashorn.api.scripting.ScriptObjectMirror;
+
+import java.util.Collection;
+import java.util.List;
+
+public class ListJavaTypeWrapper extends BasicJavaTypeWrapper {
+
+    public ListJavaTypeWrapper(Class<?> javaClass) {
+        super(javaClass);
+    }
+
+    @Override
+    boolean valid(Object value) {
+        return Collection.class.isInstance(value) ||
+                (value instanceof ScriptObjectMirror && List.class.isInstance(((ScriptObjectMirror) value).values()));
```

Comment:
BasicJavaTypeWrapper and ListJavaTypeWrapper classes are JavaScript specific type wrappers. Therefore i put them all in 'javascript' package.